src/toil_scripts/lib/jobs.py

```diff
@@ -0,0 +1,22 @@
+from toil_scripts.lib import partitions
+
+
+def sample_batcher_job(job, func, samples, *args):
```

Comment:
Nothing in here is specific to samples. I think the name should be more general.

The signature is like that of the functional `map()`. So maybe we should call this one `map_job`? And `samples` should be  called `inputs`.